src/test/java/com/github/mtakaki/credentialstorage/hibernate/UnitOfWorkApplicationListenerTest.java

```diff
@@ -0,0 +1,397 @@
+        when(this.session.beginTransaction()).thenReturn(this.transaction);
+        when(this.session.getTransaction()).thenReturn(this.transaction);
+        when(this.transaction.isActive()).thenReturn(true);
+
+        when(this.analyticsSessionFactory.openSession()).thenReturn(this.analyticsSession);
+        when(this.analyticsSession.getSessionFactory()).thenReturn(this.analyticsSessionFactory);
+        when(this.analyticsSession.beginTransaction()).thenReturn(this.analyticsTransaction);
+        when(this.analyticsSession.getTransaction()).thenReturn(this.analyticsTransaction);
+        when(this.analyticsTransaction.isActive()).thenReturn(true);
+
+        when(this.appEvent.getType()).thenReturn(ApplicationEvent.Type.INITIALIZATION_APP_FINISHED);
+        when(this.requestMethodStartEvent.getType())
+                .thenReturn(RequestEvent.Type.RESOURCE_METHOD_START);
+        when(this.responseFiltersStartEvent.getType())
+                .thenReturn(RequestEvent.Type.RESP_FILTERS_START);
+        when(this.requestMethodExceptionEvent.getType()).thenReturn(RequestEvent.Type.ON_EXCEPTION);
+        when(this.requestMethodStartEvent.getUriInfo()).thenReturn(this.uriInfo);
+        when(this.responseFiltersStartEvent.getUriInfo()).thenReturn(this.uriInfo);
+        when(this.requestMethodExceptionEvent.getUriInfo()).thenReturn(this.uriInfo);
+
+        this.prepareAppEvent("methodWithDefaultAnnotation");
+    }
+
+    @Test
+    public void opensAndClosesASession() throws Exception {
+        this.execute();
+
+        final InOrder inOrder = inOrder(this.sessionFactory, this.session);
+        inOrder.verify(this.sessionFactory).openSession();
+        inOrder.verify(this.session).close();
+    }
+
+    @Test
+    public void bindsAndUnbindsTheSessionToTheManagedContext() throws Exception {
+        doAnswer(new Answer<Object>() {
+            @Override
+            public Object answer(final InvocationOnMock invocation) throws Throwable {
+                assertThat(ManagedSessionContext
+                        .hasBind(UnitOfWorkApplicationListenerTest.this.sessionFactory))
+                                .isTrue();
+                return null;
+            }
+        }).when(this.session).beginTransaction();
+
+        this.execute();
+
+        assertThat(ManagedSessionContext.hasBind(this.sessionFactory)).isFalse();
+    }
+
+    @Test
+    public void configuresTheSessionsReadOnlyDefault() throws Exception {
+        this.prepareAppEvent("methodWithReadOnlyAnnotation");
+
+        this.execute();
+
+        verify(this.session).setDefaultReadOnly(true);
+    }
+
+    @Test
+    public void configuresTheSessionsCacheMode() throws Exception {
+        this.prepareAppEvent("methodWithCacheModeIgnoreAnnotation");
+
+        this.execute();
+
+        verify(this.session).setCacheMode(CacheMode.IGNORE);
+    }
+
+    @Test
+    public void configuresTheSessionsFlushMode() throws Exception {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://www.codacy.com/app/mitsuotakaki/CredentialStorageService-dw-hibernate/file/2455877775/issues/source?bid=3237716&fileBranchId=3250434#l142)